test/e2e/etcd_multi_node_test.go

```diff
@@ -138,6 +152,56 @@ var _ = Describe("Etcd", func() {
 	})
 })
 
+func deleteMemberDir(ctx context.Context, cl client.Client, logger logr.Logger, etcd *v1alpha1.Etcd, podName string) {
+	Expect(deleteDir(kubeconfigPath, namespace, podName, "backup-restore", "/var/etcd/data/new.etcd/member")).To(Succeed())
+	checkUnreadySts(ctx, cl, logger, etcd)
+	checkReadySts(ctx, cl, logger, etcd)
+}
+
+func deletePodAndCheckSts(ctx context.Context, cl client.Client, logger logr.Logger, etcd *v1alpha1.Etcd, podName string) {
+	pod := &corev1.Pod{}
+	ExpectWithOffset(1, cl.Get(ctx, types.NamespacedName{Name: podName, Namespace: "shoot"}, pod)).To(Succeed())
+	ExpectWithOffset(1, cl.Delete(ctx, pod, client.PropagationPolicy(metav1.DeletePropagationForeground))).To(Succeed())
+	checkUnreadySts(ctx, cl, logger, etcd)
+	checkReadySts(ctx, cl, logger, etcd)
+}
+
```

Comment:
```suggestion
func deletePodAndCheckEtcdReady(ctx context.Context, cl client.Client, logger logr.Logger, etcd *v1alpha1.Etcd, podName string) {
	pod := &corev1.Pod{}
	ExpectWithOffset(1, cl.Get(ctx, types.NamespacedName{Name: podName, Namespace: namespace}, pod)).To(Succeed())
	ExpectWithOffset(1, cl.Delete(ctx, pod, client.PropagationPolicy(metav1.DeletePropagationForeground))).To(Succeed())
	checkUnreadySts(ctx, cl, logger, etcd)
	checkEtcdReady(ctx, cl, logger, etcd, multiNodeEtcdTimeout)
}
```
Already we are checking sts is ready or not inside func `checkEtcdReady`.
So we can use straightway `checkEtcdReady`

Reply:
WDYT ?

Reply:
Makes sense
Updated